lib/galaxy/util/__init__.py

```diff
@@ -2006,3 +2007,9 @@ def lowercase_alphanum_to_hex(lowercase_alphanum: str) -> str:
     import numpy as np
 
     return np.base_repr(int(lowercase_alphanum, 36), 16).lower()
+
+
+def toContentDisposition(filename: str) -> str:
+    sanitized_filename = "".join(c in FILENAME_VALID_CHARS and c or "_" for c in filename)[0:150]
```

Comment:
You will also need to apply the file length limit and exclude certain characters in`utf8_encoded_filename` as well.

Reply:
ok, also the limit is applied to 150 characters shouldn't it be 255?

Reply:
And by `exclude certain characters` you mean which characters?

Reply:
https://github.com/galaxyproject/galaxy/pull/19161#discussion_r1850153897

Reply:
I couldn't find any specific limitation on characters from the link but I found out that for a typical filename, the characters limitation is 256 characters and `/\?%*:|"<>` are not allowed.

Reply:
Sounds good. I bet most browsers are putting proper restrictions in place anyway.